tock/tock/static/js/components/timecard.js

```diff
@@ -363,6 +368,56 @@ function handleConfirm(val) {
   }
 }
 
+/** @function
+ * Show or hide the billing hour summation elements in the footer
+ * @name handleBillingElementState
+ * @param {string} visibility
+ * */
+function handleBillingElementState(visibility) {
+  const showBillingHourSummationElements = visibility === 'show';
+  const totalReportedElement = document.querySelector('#total-reported-div');
+  const totalBillableElement = document.querySelector('#total-billable-div');
+
+  if (showBillingHourSummationElements) {
+    const weeklyBillingAlertElement = document.querySelector('#weekly-billing-alert');
+    const addItemWrapper = document.querySelector('#add-item-wrapper');
+    addItemWrapper.classList.remove('grid-col-2');
+    addItemWrapper.classList.add('grid-col-8');
+    weeklyBillingAlertElement.classList.add('entry-hidden');
+    totalReportedElement.classList.remove('entry-hidden');
+    totalBillableElement.classList.remove('entry-hidden');
+  } else {
+    const weeklyBillingAlertElement = document.querySelector('#weekly-billing-alert');
+    const addItemWrapper = document.querySelector('#add-item-wrapper');
+    addItemWrapper.classList.remove('grid-col-8');
+    addItemWrapper.classList.add('grid-col-2');
+    weeklyBillingAlertElement.classList.remove('entry-hidden');
+    totalReportedElement.classList.add('entry-hidden');
+    totalBillableElement.classList.add('entry-hidden');
+  }
+}
+
+/** @function
+ * Examine all of the timecard entries and determine if any of them are weekly billing
+ * @name weeklyBilledProjectExists
+ * */
+function weeklyBilledProjectExists() {
+  const selects = document.querySelectorAll('.entry-project select');
+  let weeklyBilledProjectSeen = false;
+  for (let i = 0; i < selects.length; i++) {
+    const currentSelect = selects[i];
+    const project = currentSelect.selectedOptions[0];
+    const projectIsWeeklyBilled = (project.dataset || {}).is_weekly_bill === 'true';
+    // Flip our 'state' so we know we've encountered at least one weekly billed project
+    // Get out of the loop as soon as we've found one weekly billed project
+    if (projectIsWeeklyBilled && !weeklyBilledProjectSeen) {
+      weeklyBilledProjectSeen = true;
```

Comment:
**suggestion (non-blocking):** I think that we could `return true;` directly from here and then change L418 to `return false;`. Then the `weeklyBilledProjectSeen` variable isn't needed at all.